homestar-runtime/src/network/swarm.rs

```diff
@@ -53,15 +57,47 @@ pub(crate) async fn new(settings: &settings::Node) -> Result<Swarm<ComposedBehav
     )
     .build();
 
+    startup(&mut swarm, &settings.network)?;
+
+    Ok(swarm)
+}
+
+fn startup(swarm: &mut Swarm<ComposedBehaviour>, settings: &settings::Network) -> Result<()> {
```

Comment:
Here's a possible cleanup of this bit, particularly b/c we don't use the result of `.ok()` and that inner match is unnecessary: 

```suggestion
fn startup(swarm: &mut Swarm<ComposedBehaviour>, settings: &settings::Network) -> Result<()> {
    // Listen-on given address
    swarm.listen_on(settings.listen_address.to_string().parse()?)?;

    // Dial trusted nodes specified in settings. Failure here shouldn't halt node startup.
    settings.trusted_node_addresses.iter().for_each(|addr| {
        let _ = swarm
            .dial(addr.clone())
            .map(|_| info!(addr=?addr, "successfully dialed configured node"))
            // log dial failure and continue
            .map_err(|e| error!(err=?e, "failed to dial node"));

        // add node to kademlia routing table
        if let Some(Protocol::P2p(peer_id)) = addr.iter().find(|i| matches!(i, Protocol::P2p(_))) {
            info!(addr=?addr, "added configured node to kademlia routing table");
            swarm
                .behaviour_mut()
                .kademlia
                .add_address(&peer_id, addr.clone());
        } else {
            warn!(addr=?addr, err="node address doesn't include peer ID",
                  "node not added to kademlia routing table")
        }
    });

    // join on `receipts` topic
    swarm
        .behaviour_mut()
        .gossip_subscribe(pubsub::RECEIPTS_TOPIC)?;

    Ok(())
}
```

I also trimmed the logs here b/c of assumptions on trusted/configured being the same thing.

Reply:
we dont use the result of dial either (just log) so the `.ok()` skips the compiler warning for not using a result, though really a `match` here makes more sense anyways.

Reply:
@mriise Yeah, totally get that we don't use the result. Typically, `let _` is preferred over `.ok` as it's more explicit that we're not caring about it all, vs having an Option of None.

Reply:
Nonetheless, a `match` variant with the error logged is probably the way forward, to your point.